src/main/java/org/terasology/logic/health/RegenAuthoritySystem.java

```diff
@@ -124,7 +124,9 @@ private void regenerate(long currentTime) {
                 health.nextRegenTick = currentTime + 1000;
                 if (health.currentHealth >= health.maxHealth) {
                     regenToBeRemoved.add(regen.getLowestEndTime());
-                    entity.removeComponent(RegenComponent.class);
+                    if (regen.hasBaseRegenOnly()) {
+                        entity.removeComponent(RegenComponent.class);
+                    }
                     entity.send(new OnFullyHealedEvent(entity));
```

Comment:
This will not lead to multiple `OnFullyHealedEvent` as the entity is removed from the scheduler on reaching full health ([Line 126](https://github.com/Terasology/Health/pull/13/files#diff-7de79d955d70c98aa3d1ff0a21c48faeR126))